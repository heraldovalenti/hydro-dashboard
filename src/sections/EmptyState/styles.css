.empty-state {
  width: 560px;
  position: absolute;
  right: 0;
  left: 0;
  margin: auto;
  top: 24vh;
  font-family: 'Open Sans';
  text-align: center;
  padding-top: 48px;
  padding-bottom: 48px;
}

.empty-state__title {
  font-size: 36px;
  font-weight: 600;
}

.empty-state__subtitle {
  font-size: 18px;
}
