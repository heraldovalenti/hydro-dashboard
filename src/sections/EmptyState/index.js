import React from 'react';
import { Button } from '@material-ui/core';

import './styles.css';

export default function EmptyState(props) {
  return (
    <div className="empty-state">
      {props.icon && <div className="empty-state__icon">{props.icon}</div>}
      {props.image && <div className="empty-state__image">{props.image}</div>}
      {props.title && <h1 className="empty-state__title">{props.title}</h1>}
      {props.subtitle && (
        <p className="empty-state__subtitle">{props.subtitle}</p>
      )}
      <div className="empty-state__actions">
        {props.primaryActions.map((action) => (
          <Button {...action}>{action.text}</Button>
        ))}
        {props.secondaryActions.map((action) => (
          <Button {...action}>{action.text}</Button>
        ))}
      </div>
    </div>
  );
}

EmptyState.defaultProps = {
  primaryActions: [],
  secondaryActions: [],
};
