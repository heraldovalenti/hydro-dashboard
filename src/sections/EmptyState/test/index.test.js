import React from 'react';
import EmptyState from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('<EmptyState/>', () => {
    it('default', () => {
        const props = {
            icon: 'icon',
            image: 'image',
            title: 'empty title',
            subtitle: 'empty subtitle',
            primaryActions: [{text: 'action 1'}],
            secondaryActions: [{text: 'action 2'}],
        }
        const renderer = new ShallowRenderer();
        renderer.render(<EmptyState {...props}/>);
        const tree = renderer.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });
});