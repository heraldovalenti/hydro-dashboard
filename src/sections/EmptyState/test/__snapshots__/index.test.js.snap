// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState/> default 1`] = `
<div
  className="empty-state"
>
  <div
    className="empty-state__icon"
  >
    icon
  </div>
  <div
    className="empty-state__image"
  >
    image
  </div>
  <h1
    className="empty-state__title"
  >
    empty title
  </h1>
  <p
    className="empty-state__subtitle"
  >
    empty subtitle
  </p>
  <div
    className="empty-state__actions"
  >
    <ForwardRef(WithStyles)
      text="action 1"
    >
      action 1
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      text="action 2"
    >
      action 2
    </ForwardRef(WithStyles)>
  </div>
</div>
`;
